main.py

```diff
@@ -1557,20 +1605,30 @@ def format_completion(self, item) -> 'Tuple[str, str]':
             insertText = item.get("insertText")
         if insertText[0] == '$':  # sublime needs leading '$' escaped.
             insertText = '\$' + insertText[1:]
-        return ("{}\t{}".format(label, detail), insertText)
+        return "{}\t{}".format(label, detail), insertText
 
     def handle_response(self, response):
-        items = response["items"] if isinstance(response,
-                                                dict) else response
-        self.completions = list(self.format_completion(item) for item in items)
-        self.run_auto_complete()
+        if self.state == CompletionState.REQUESTING:
+            items = response["items"] if isinstance(response,
+                                                    dict) else response
+            names = list(item.get('label') for item in items)
+            debug('got completions', names)
+            self.completions = list(self.format_completion(item) for item in items)
```

Comment:
Will you be doing anything with the `sortText` property of the CompletionItem or will you let ST sort the results? The `clangd` server uses `sortText` to sort completions by priority, so you'll see properties and methods first, base classes somewhere middle, keywords last.

Reply:
Interesting, is there a way to control what ST does with the completions?

Reply:
No there is not. ST merges and sorts all items it gets internally.

Reply:
Not immediately, it just presents that list as-is if the user hasn't typed more characters yet, so there might be some benefit to using `sortText`.

Reply:
But it could slow down LSP a lot if there are tons of completion items, you'll have to experiment with it.

Reply:
For reference: https://github.com/llvm-mirror/clang-tools-extra/blob/c543235c134fbb02fb630eaa53661b188068b267/clangd/ClangdUnit.cpp#L316-L317

Reply:
@rwols good to know!